testcases/kernel/device-drivers/common/common.sh

```diff
@@ -0,0 +1,425 @@
+    ;;
+    *)
+      die "check_kernel_errors: No logic for type $type yet."
+    ;;
+  esac
+}
+
+# Function to check if environment variable is set
+# Input: $1 - environment variable to be checked
+# returns true if set
+# returns with error if not set
+check_env_var() {
+  output_str=`env|grep $1`
+  if [ ${#output_str} == 0 ]
+  then
+    die "$1 not defined"
+  fi
+}
+
+# $1: check type, either 'y', 'm', 'ym' or 'n'
+# $2: Options to check. Uses same syntax returned by get_modular_config.names.sh
+#     which is CONFIG1^CONFIG2:module1 CONFIG3:module2
+check_config_options()
+{
+  config_cmd='zcat /proc/config.gz'
+  ls /boot/config-`uname -r` &> /dev/null && config_cmd="cat /boot/config-"`uname -r`
+  case $1 in
+    y) check='=y';;
+    m) check='=m';;
+    ym) check='(=y|=m)';;
+    n) check=' is not set';;
+    *) die "$1 is not a valid check_config_options() option"
+  esac
+  OIFS=$IFS
+  IFS=' '
+  shift
+  x=$*
+  x=${x[@]}
+  y=()
+  for i in $x
+  do
+    newval=`echo $i | cut -d':' -f 1`
+    y+=($newval)
+  done
+  IFS='^';y=${y[@]}
+  IFS=' '
+  for option in $y; do
+     $config_cmd | egrep "$option$check" && continue ||  (IFS=$OIFS; \
+     if is_opt_config "$option"; then skip_test "optional $option not enabled"; else die "$option is not $check"; fi); exit $?
+  done
+  IFS=$OIFS
+}
+
+# To get instance number from dev node
+# Input:
+#   $1: dev node like /dev/rtc0, /dev/mmcblk0, /dev/sda1, /dev/mtdblk12 etc
+# Output:
+#   instance number like '0', '1' etc
+get_devnode_instance_num()
+{
+  devnode_entry=$1
+  inst_num=`echo $devnode_entry |grep -oE "[[:digit:]]+$" ` || die "Failed to get instance number for dev node entry "$devnode_entry" "
+  echo $inst_num
+}
+
+# Get filesize
+#   $1: filename
+#   return: file size in byte
+get_filesize()
+{
+  inputfile=$1
+  fs=`wc -c < $inputfile`
+  echo $fs
+}
+
+# hexdump one byte at offset $oset from $filename
+#   $1: filename
+#   $2: offset
+hexdump_onebyte()
+{
+  local filename=$1
+  local offset=$2
+  local byte=`hexdump -n 1 -s $offset -C $filename | head -1 | awk -F " " '{print $2}'`
+  echo $byte
+}
+
+# replace one byte of inputfile
+# Input
+#   $1: inputfile
+#   $2: offset - decimal number and starting from 0
+#   $3: new_byte - need to be hex
+replace_onebyte()
+{
+  local inputfile=$1
+  local offset=$2
+  local new_byte=$3
+
+  #local fs=`wc -c < $inputfile`
+  local fs=`get_filesize $inputfile`
+  echo "$inputfile size is: $fs"
+  tempfile="$TMPDIR/tempfile_to_replace_$$"
+  do_cmd "dd if=$inputfile of=$tempfile bs=1 count=$offset"
+  test_print_trc "echo -ne "\x$new_byte" >> $tempfile"
+  echo -ne "\x$new_byte" >> $tempfile
+  do_cmd "dd if=$inputfile of=$tempfile bs=1 count=$(( $fs - $offset - 1 )) skip=$(( $offset + 1 )) seek=$(( $offset + 1 ))"
+
+  do_cmd "cp $tempfile $inputfile"
+}
+
+# wrapper for wget
+Wget()
+{
+  wget $* || wget --proxy off $* || http_proxy=$SITE_HTTP_PROXY wget $*
+}
```

Comment:
This is completely non-relevant to the test itself. Also anything that downloads code at runtime is no-go for LTP.

Reply:
We do have use cases where it is not feasible to have all the files pre-installed in the filesystem. For example, many of the video tests use big test files, and it is not practical and sometimes even possible to have all of them installed in the rootfs. For such cases, the test logic downloads the multimedia file on the fly as part of the test logic.

Reply:
That would be another tough problem to crack, some systems don't even have access to network at all.

But more importantly, can we trim anything that is not used by the actual test so that we start with a reasonably small patch for the start? We can add things as they are needed later on.